mlflow/genai/judges/instructions_judge/__init__.py

```diff
@@ -184,6 +195,21 @@ def __call__(
                 error_code=INVALID_PARAMETER_VALUE,
             )
 
+        # Extract fields from trace if needed (when not using {{ trace }} template)
```

Comment:
@BenWilson2 This function is getting pretty long. Can we split it into a few helper functions, e.g. for validation, trace-base evaluation, non-trace-based evaluation (or whatever split is intuitive to you)?